### DISCUSSION TITLE
RPI os + software environment + application up to date deployment method

### DISCUSSION THREAD

**Discussion by lida2003** [thecognifly/cognifly-python#3] [Ideas]
What I'm suggesting here is kind of manual and script, which can keep the RPI system up to date.

As we know:
a) raspberry pi os is quite stable
b) apt repo is also stable (which I think)
c) application and library release, we consider to be safe to update

Now, we comes back to our discussion "RPI os + software environment + application up to date deployment method"
a) raspbian os is installed by it's scripts or tools  // guide needed
b) software environment update // script needed
c) application & library update  // script or ota tool needed

Is there any plan for cognifly to write these scripts and manuals or corresponding tools?

**Comment by yannbouteiller** [comment]
Not at the moment. The library is a python library and can be updated via pip, whereas the Flight Controller framework is from INAV and needs to be updated via inav-configurator.

cognifly-python works on raspberry pi but also on google coral, so it is up to the user to setup the OS that they wish to use. We just provide readily setup raspbian images for convenience.

**Reply by lida2003** [reply]
a)  Flight Controller framework and cognifly configuration are flight control firmware of inav and configuration, right?
==> And this configuration is for a specific quard, which is a F4 contoller.
```
# version
# INAV/BETAFLIGHTF4 2.3.0 Jul  9 2021 / 21:57:12 (cd15ba9b7)
# GCC-8.2.1 20181213 (release) [gcc-8-branch revision 267074]
```
b) Is there any key points of configuration that should be pay special attention to such as arm/disarm channel, throttle for hover, etc.  And suggested value for F4 (well,  cognify frame) uses a set of parameters. Other quards should be carefully tuned. Some thing like [matek module 3901 guide](http://www.mateksys.com/downloads/setup_optic%20flow_in_INAV_rev1.pdf)

**Reply by yannbouteiller** [reply]
Most configuration parameters (in particular those not related to the drone's physics such as channels configuration) can remain as they are in the provided configuration .txt file.

Those that need special attention if using another quad are those related to the physics, such as the PID values.

Note however that this is with regard to the version that is currently supported on PyPI, the config .txt file that we use for inav6 on the `yb/stable` branch is not yet available. I can provide it if you need, though.

**Reply by lida2003** [reply]
Thank you. I hope Google AI Kit can arrive as soon as possible.
